src/main/java/com/fauna/codec/FaunaTokenType.java

```diff
@@ -44,14 +44,14 @@ public enum FaunaTokenType {
 
     MODULE;
 
-    public FaunaTokenType getEndToken() throws IOException {
+    public FaunaTokenType getEndToken() {
         switch (this) {
             case START_DOCUMENT: return END_DOCUMENT;
             case START_OBJECT: return END_OBJECT;
             case START_ARRAY: return END_ARRAY;
             case START_PAGE: return END_PAGE;
             case START_REF: return END_REF;
-            default: throw new IllegalStateException("No end token for " + this.name());
+            default: throw new ClientResponseException("No end token for " + this.name());
```

Comment:
Not sure I agree with this one because it's in codec and not specifically response, but we can think about that more deeply another time.